### ISSUE TITLE
PAPI Displaying Hex Color Codes Incorrectly

### ISSUE THREAD

**Issue by RoboTitan** [PlaceholderAPI/PlaceholderAPI#343]
[Request change (Wiki)]: https://github.com/PlaceholderAPI/PlaceholderAPI/issues/new?template=change_request_wiki.md
[Request change (PlaceholderAPI)]: https://github.com/PlaceholderAPI/PlaceholderAPI/issues/new?template=change_request_placeholderapi.md
[Spigot]: https://www.spigotmc.org/resources/6245/
[Jenkins]: http://ci.extendedclip.com/job/PlaceholderAPI/
[issues]: https://github.com/PlaceholderAPI/PlaceholderAPI/issues

## Please read
This template is only for reporting bugs of PlaceholderAPI!  
If you want to request changes to the code of PlaceholderAPI, use the [Request change (PlaceholderAPI)] template.  
For changes to the wiki use the [Request change (Wiki)] template.

Please also make sure that you use the [latest release][Spigot] or the latest [developement build][Jenkins] and that your bug isn't already reported on the [issues] page.

**Please report issues with expansions/placeholders to the corresponding issue tracker! Only report it here if you can't find any place to report it!**

### Issue
> What is the issue? Describe it like you would tell a friend.

Hex color codes in placeholders from 1.16 do not properly translate in PAPI

### Expected behaviour
> What should PlaceholderAPI do?

PAPI should take the entire string of the color code, translating it into the correct color

### Actual behaviour
> What does PlaceholderAPI actually do?

PAPI takes the last number or letter of the hex string and uses that as the color code
![PAPI-Bug](https://user-images.githubusercontent.com/67801380/86490300-e79d0d80-bd34-11ea-8fff-b290d5488ec3.png)

### How to reproduce
> What steps did you made, to get this bug?

1. Use a plugin that supports hex color (such as essentials)
2. If using essentials and essentials chat, create a nickname with hex codes
     example" /nick &#e8c3cdT&#2dc621e&#7089dfs&#ff8b00t
3. This should appear as "Test: with "T" in a light pink color, "e" in a green color, "s" in a grayish blue color, and "t" in an orange   color
4. Use a placeholder that uses the display name suck as"%player_displayname%" or "%essentials_nickname%"
5. Rather than using the entire hex code, PAPI will display with "&dT&1e&fs&0t"

**Comment by Andre601** [comment]
PlaceholderAPI does nothing more than make a `translateAlternateColorCodes('&', text)` on the placeholders which only translates the provided character (&) with the next following one (a-f, 0-9) or something like that.
This isn't really a PlaceholderAPI issue as the plugin does nothing special in terms of translating color codes.
Additionally is the plugin not fully 1.16 as a PR (#333) still needs to be merged.

So tl;dr is, that this is nothing that really needs fixing as it is nothing that PAPI does specifically.

**Comment by extendedclip** [comment]
The plugin handling the messages should handle the hex codes after PAPI provides the parsed string. I don't see a need to add direct support for hex in papi.
